src/Lean/Parser/Extra.lean

```diff
@@ -57,6 +57,12 @@ attribute [runBuiltinParserAttributeHooks]
 @[runBuiltinParserAttributeHooks, inline] def manyIndent (p : Parser) : Parser :=
   withPosition $ many (checkColGe "irrelevant" >> p)
 
+@[runBuiltinParserAttributeHooks, inline] def sepByIndent (p : Parser) (sep : String) (psep : Parser := symbol sep) (allowTrailingSep : Bool := false) : Parser :=
+  withPosition $ sepBy (checkColGe "irrelevant" >> p) sep psep allowTrailingSep
```

Comment:
I wonder if `sepByIndent` should automatically accept newlines as separator.  From what I can tell, this is the case everywhere it's used.

```suggestion
  withPosition $ sepBy (checkColGe "irrelevant" >> p) sep (psep <|> checkLinebreakBefore >> pushNone) allowTrailingSep
```